library/kotlin/src/io/envoyproxy/envoymobile/Client.kt

```diff
@@ -0,0 +1,73 @@
+package io.envoyproxy.envoymobile
+
+import java.nio.ByteBuffer
+
+interface Client {
+  fun startStream(request: Request, streamCallback: StreamCallback): StreamEmitter
+}
+
+interface StreamCallback {
+  fun onHeaders(statusCode: Int, headers: Map<String, List<String>>)
+  fun onData(byteBuffer: ByteBuffer, endStream: Boolean)
+  fun onMetadata(metadata: Map<String, List<String>>, endStream: Boolean)
+  fun onTrailers(trailers: Map<String, List<String>>)
+  fun onError(error: Error)
+
+  fun onCanceled() // TODO: onInterrupt()?
+  fun onCompletion()
+
+}
+
+interface StreamEmitter {
+  fun sendHeaders(headers: Map<String, List<String>>): Boolean
+  fun sendData(byteBuffer: ByteBuffer, endStream: Boolean): Boolean
+  fun sendMetadata(metadata: Map<String, List<String>>, endStream: Boolean): Boolean
+  fun sendTrailers(trailers: Map<String, List<String>>): Boolean
+
+  fun close(): Boolean
```

Comment:
Per offline discussion: Remove since we probably don't need this yet. Semantically, this is essentially setting `endStream=true` in one of the above